### ISSUE TITLE
andOTP as 2FA app in Uber

### ISSUE THREAD

**Issue by jacek-1** [andOTP/andOTP#324]
Hi,

I use andOTP on Samsung A5 with android 7, I turned on 2FA settings on Uber app, besides andOTP I have also _symantec vip access_ and _MS authenticator_ apps installed, both were listed as choices to add when setting up 2FA on Uber, but not andOTP,

i'm checking on my web uber profile https://m.uber.com/looking but it looks like 2FA can be set up only on a mobile app

any workaround for that ?

thanks

**Comment by flocke** [comment]
Let me look into the 2FA on Uber if I find the time. It looks like they call an Android Intent to set up 2FA directly on the phone. In that case andOTP does not (yet) support this function. But if I find out the exact command they use I will try and add support for it.

**Comment by jacek-1** [comment]
hi again, 
any chance to tackle that issue ?

**Comment by flocke** [comment]
Sorry, I didn't have a chance to look at this yet. I currently don't have any time to work on andOTP, but I hope this will change soon.

**Comment by jacek-1** [comment]
that's fine, no pressure,

**Comment by mbertram** [comment]
The URI format used by Google Authenticator is described [here](https://github.com/google/google-authenticator/wiki/Key-Uri-Format).

```xml
<intent-filter>
   <action android:name="android.intent.action.VIEW" />
   <category android:name="android.intent.category.DEFAULT" />
   <category android:name="android.intent.category.BROWSABLE" />
   <data android:scheme="otpauth" android:host="totp" />
   <data android:scheme="otpauth" android:host="hotp" />
</intent-filter>
```

**Comment by moritzgloeckl** [comment]
I just looked into this, it's actually possible to setup andOTP with Uber. Here's how I did it (the buttons may be named a bit different, these are my own translations):

1. Open Uber App on Android
2. "Settings"
3. Scroll down to "Security"
4. Tap "Two-Step Authentication"
5. Tap "Setup now"
6. Enter your password
7. Choose "Security-App"
8. Now on my device it will open the Google Authenticator to store the Key
9. Tap "Cancel"
10. Go back using the Android hardware back button
11. Tap "I need help"
12. Tap "Set up another device"
13. It will show the QR-Code and the security key which can be copied and added to andOTP

Working great :)

**Comment by jacek-1** [comment]
@moritzgloeckl, it works, thanks for this hint!

**Comment by flocke** [comment]
Nice, thank you for figuring this out. I will keep this issue open so I remember to look into the other method. But since there is a method that works I'll push the priority back a bit.

**Comment by schwedenmut** [comment]
@flocke if you want, I can take a look at this to make andOTP receive otpauth data from other applications.

**Comment by flocke** [comment]
Sure, feel free to look into it. I always keep pushing this back because I don't use any app that does this myself and then I tend to forget about it.

**Comment by flocke** [comment]
Implemented in PR #393 by @schwedenmut
